tests/conftest.py

```diff
@@ -0,0 +1,39 @@
+"""Global test configuration."""
+import json
+import os
+
+import pytest
+import requests
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup() -> None:  # noqa: PT004
+    """Sets up testing environment by downloading demofiles."""
+    with open("tests/test_data.json", encoding="utf-8") as f:
+        demo_data = json.load(f)
+    for file in demo_data:
+        _get_demofile(demo_link=demo_data[file]["url"], demo_name=file)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def teardown() -> None:  # noqa: PT004
+    """Cleans testing environment by deleting all .dem and .json files."""
+    files_in_directory = os.listdir()
```

Comment:
I think you need to add a `yield` here so that the tests are run before this part. Otherwise i think the function will run instantly and just delete the downloaded files?

Reply:
Changing the scope of the two fixtures back to `session` and adding a `yield` to the start of the teardown gets it to work for me locally.

But even with this it will still get downloaded for every build in the matrix and every new PR/commit.